mrp_client.py

```diff
@@ -95,26 +124,47 @@ def preseed(self, command, preseed_name, preseed_file, preseed_desc,
     parser_preseed.add_argument('--action', type=str, choices=['check', 'upload'],
                                 required=True, default='')
     parser_preseed.add_argument('--preseed-name', type=str, default='',
                                 required=True, help='Name of the preseed in MrP')
     parser_preseed.add_argument('--preseed-path', type=str, default='',
-                                help='Path to the preseed file')
+                                required=False, help='Path to the preseed file')
     parser_preseed.add_argument('--type', type=str, choices=['kickstart', 'preseed'],
-                                default=None, help='Type of the preseed file')
+                                required=False, default=None, help='Type of the preseed file')
     parser_preseed.add_argument('--description', type=str, default='',
-                                help='Description of the preseed file in MrP')
+                                required=False, help='Description of the preseed file in MrP')
     parser_preseed.add_argument('--knowngood', action='store_true', default=False,
-                                help='Switches known good flag')
+                                required=False, help='Switches known good flag')
     parser_preseed.add_argument('--public', action='store_true', default=False,
-                                help='Switches public flag')
+                                required=False, help='Switches public flag')
+
+    parser_machine = subparsers.add_parser('state')
+    parser_machine.add_argument('--action', type=str, default='', required=True,
+                                choices=['provision', 'setparams', 'getparams'],
+                                help='provision, setparams, getparams')
+    parser_machine.add_argument('--machine', type=str, default='',
+                                required=True, help='name of the machine')
+    parser_machine.add_argument('--preseed-name', type=str, default='',
+                                required=False, help='name of the preseed to use')
+    parser_machine.add_argument('--initrd-desc', type=str, default='',
+                                required=False, help='description of the initrd to use')
+    parser_machine.add_argument('--kernel-desc', type=str, default='',
+                                required=False, help='description of the kernel to use')
+    parser_machine.add_argument('--kernel-opts', type=str, default='',
+                                required=False, help='kernel options to use')
+    parser_machine.add_argument('--arch', type=str, default='', required=False,
+                                help='architecture of the machine as in MrP')
+    parser_machine.add_argument('--subarch', type=str, default='',
+                                required=False, help='subarchitecture of the machine as in MrP')
+    parser_machine.add_argument('--netboot', action='store_true', default=None, required=False,
```

Comment:
This needs to be type=bool, default=None, so that we can pass --netboot=True or False.

Right now, there are only two choices: None (don't change) and True (enable it), ie, it's impossible to disable.

Reply:
type=bool is broken in argparse has it is doing bool(stdin) which invariably leads (lest there is nothing) it to be True.

To keep it consistent with the other type=bool I've added a --netboot-disabled, but can change it to a type=str and cast the result.